src/Serilog.Sinks.Loggly/Sinks/Loggly/FileBufferDataProvider.cs

```diff
@@ -0,0 +1,279 @@
+        }
+
+        public IEnumerable<LogglyEvent> GetBatchOfEvents()
+        {
+            //if current batch has not yet been processed, return it
+            if (_currentBatchOfEventsToProcess != null)
+                return _currentBatchOfEventsToProcess;
+
+            //if we have a bookmark in place, it may be the next position to read from
+            // otherwise try to get a valid one
+            if (_currentBookmark == null)
+            {
+                //read the current bookmark from file, and if invalid, try to create a valid one
+                _currentBookmark = TryGetValidBookmark();
+
+                if (!IsValidBookmark(_currentBookmark))
+                    return new List<LogglyEvent>();
+            }
+
+            //bookmark is valid, so lets get the next batch from the files.
+            _currentBatchOfEventsToProcess = GetListOfEvents(_currentBookmark.FileName);
+            //_futureBookmark = new Bookmark(nextByteCount, _currentBookmark.FileName); //performed in futureBookmark
+            return _currentBatchOfEventsToProcess;
+        }
+
+        public void MarkCurrentBatchAsProcessed()
+        {
+            //reset internal state: 
+            if(_futureBookmark != null)
+                _bookmarkProvider.UpdateBookmark(_futureBookmark);
+
+            //we can move the marker to what's in "future" (next expected position)
+            _currentBookmark = _futureBookmark;
+            _currentBatchOfEventsToProcess = null;
+        }
+
+        public void MoveBookmarkFoward()
+        {
+            // Only advance the bookmark if no other process has the
+            // current file locked, and its length is as we found it.
+            var fileSet = GetEventBufferFileSet();
+            if (fileSet.Length == 2 
+                && fileSet.First() == _currentBookmark.FileName 
+                && IsUnlockedAtLength(_currentBookmark.FileName, _currentBookmark.Position))
+            {
+                //move to next file
+                _bookmarkProvider.UpdateBookmark(new Bookmark(0, fileSet[1]));
+            }
+
+            if (fileSet.Length > 2)
+            {
+                // Once there's a third file waiting to ship, we do our
+                // best to move on, though a lock on the current file
+                // will delay this.
+                _fileSystemAdapter.DeleteFile(fileSet[0]);
+            }
+        }
+
+        bool IsUnlockedAtLength(string file, long maxLen)
+        {
+            try
+            {
+                using (var fileStream = _fileSystemAdapter.Open(file, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.Read))
+                {
+                    return fileStream.Length <= maxLen;
+                }
+            }
+#if HRESULTS
+            catch (IOException ex)
+            {
+                var errorCode = Marshal.GetHRForException(ex) & ((1 << 16) - 1);
+                if (errorCode != 32 && errorCode != 33)
```

Comment:
It would be nice if you could you put those values inside a const that explains what error they represent

Reply:
I'll have to ask around what this means since It was in the original code and I've never had to go through that. I'll look into it. Something more explicit is definitely more beneficial.

Reply:
(y)